pkg/clientconn/client_test.go

```diff
@@ -138,3 +140,34 @@ func (t *ClientTestSuite) TestAuthenticatedHTTPTransport_WebSocket() {
 		})
 	}
 }
+
+func (t *ClientTestSuite) TestGRPCShouldRetry() {
+	testCases := map[string]int{
+		"retry on too large message": 3,
```

Comment:
nit: do we expect it to be not that large second time? Might be something like "try to turn it off and on again"?